processor/spanmetricsprocessor/processor.go

```diff
@@ -73,6 +74,8 @@ type processorImp struct {
 	latencySum          map[metricKey]float64
 	latencyBucketCounts map[metricKey][]uint64
 	latencyBounds       []float64
+	latencyExemplarTraceIDs map[metricKey][]pdata.TraceID
+	latencyExemplarValues   map[metricKey][]float64
```

Comment:
If there's a 1:1 mapping between `TraceID` and `Values`, I wonder if it would make sense to:
- Declare a new `type  exemplars map[pdata.TraceID]float64`
- Then here we can do: `latencyExemplars map[metricKey]exemplars`

It should then simplify things further down; e.g. we won't need to initialise, set and read two different slices.

Reply:
@albertteoh Hello, good idea and thx for the suggestion! I would also like to add a little suggestion to simplify and keep the bucket index and to have a unique structure like:

```
type exemplarData struct {
    traceID pdata.TraceID
    value float64
}
...
latencyExemplarDatas   map[metricKey][]exemplarData
```

it needs to be a slice of exemplar Data in order to keep track of the bucket index.

What do you think ?

Reply:
Ah, yes you're right, I missed that. Makes sense to have an `exemplarData` struct 👍🏼

Reply:
@albertteoh done